secrets/secrets-vault/src/lib.rs

```diff
@@ -0,0 +1,497 @@
+use anyhow::{anyhow, Result};
+use async_nats::{HeaderMap, Message, Subject};
+use ed25519_dalek::pkcs8::EncodePrivateKey;
+use futures::StreamExt;
+use jsonwebtoken::{get_current_timestamp, Algorithm, EncodingKey};
+use nkeys::{KeyPair, XKey};
+use serde::{Deserialize, Serialize};
+use std::{net::SocketAddrV4, result::Result as StdResult, str::FromStr};
+use tracing::debug;
+use vaultrs::{
+    api::kv2::{requests::ReadSecretRequest, responses::ReadSecretResponse},
+    client::{Client, VaultClient, VaultClientSettingsBuilder},
+};
+use wascap::{
+    jwt::{CapabilityProvider, Component, Host},
+    prelude::Claims,
+};
+use wasmcloud_secrets_types::{
+    GetSecretError, Secret, SecretRequest, SecretResponse, RESPONSE_XKEY, WASMCLOUD_HOST_XKEY,
+};
+
+mod jwk;
+mod jwks;
+use jwks::VaultSecretsJwksServer;
+
+const SECRETS_API_VERSION: &str = "v0";
+
+#[derive(Debug, PartialEq)]
+pub(crate) enum Operation {
+    Get,
+    ServerXkey,
+}
+
+impl FromStr for Operation {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "get" => Ok(Self::Get),
+            "server_xkey" => Ok(Self::ServerXkey),
+            subject => {
+                anyhow::bail!("unsupported subject: {subject:?}")
+            }
+        }
+    }
+}
+
+struct RequestClaims {
+    host: Claims<Host>,
+    component: Option<Claims<Component>>,
+    provider: Option<Claims<CapabilityProvider>>,
+}
+
+impl TryFrom<&SecretRequest> for RequestClaims {
+    type Error = anyhow::Error;
+
+    fn try_from(request: &SecretRequest) -> StdResult<Self, Self::Error> {
+        let host = Claims::<Host>::decode(&request.context.host_jwt)
+            .map_err(|_| anyhow!("failed to decode host claims"))?;
+
+        let component = Claims::<Component>::decode(&request.context.entity_jwt);
+        let provider = Claims::<CapabilityProvider>::decode(&request.context.entity_jwt);
+
+        if component.is_err() && provider.is_err() {
+            return Err(anyhow!("failed to decode component and provider claims"));
```

Comment:
```suggestion
            return Err(anyhow!("failed to decode entity_jwt as component or provider claims"));
```

Just nit so the expectation isn't that they both match